### ISSUE TITLE
Default environment should not be dev

### ISSUE THREAD

**Issue by toymachiner62** [kunklejr/node-db-migrate#219]
I don't think the environment should default to dev because you can accidently populate your dev db if you forget to specify the env paramater.

An example use case is we have 3 servers - dev, int, and prod, which have those same NODE_ENV's set (so the NODE_ENV=dev on the dev server etc.).

All the developers also have their own local mongo instances installed as well. If I just run `$ db-migrate up` on my local machine, it's running those migrations against the dev server even though I just want them to run locally.

I'm advocating that the default environment should be "undefined" because that is what the NODE_ENV is by default.

Then in my `database.json` file i would just have a block for the "undefined" environment which is actually my local environment.

```json
{
    "undefined": {
        "driver": "mongodb",
	"host": "localhost",
	"database": "ease"
    },
}
```

Thoughts?

**Comment by wzrdtales** [comment]
Well, I always used dev for my local database and something like test_environment for the test servers. Because dev means develop to me, so this is definitely a behavior that changes from user to user.

We can default to something else than dev, but I would suggest "default" instead of "undefined". Undefined would be misleading again and nothing were won through this change.

**Comment by toymachiner62** [comment]
The only reason i suggested undefined is because it is the default NODE_ENV. Otherwise a user will always have to specify their environment with the `-e` option.

**Comment by toymachiner62** [comment]
I do agree though that undefined is a confusing env name.

**Comment by toymachiner62** [comment]
> Undefined would be misleading again and nothing were won through this change.

The thing you'd win with this change is that users couldn't accidently populate their "dev" database due to not specifying an environment with the "-e" option.

**Comment by wzrdtales** [comment]
Yeah, I don't say you're wrong, but I feel like the most devs think that dev is their local database.
> Because dev means develop to me, so this is definitely a behavior that changes from user to user.

That leads my to one thought, might this change be a breaking change? Because every user currently using db-migrate and don't have the NODE_ENV set, would have to update his configs.

**Comment by toymachiner62** [comment]
Yes you're right i think it would be a breaking change.

**Comment by wzrdtales** [comment]
Ok there are news:
default is already considered https://github.com/kunklejr/node-db-migrate/blob/master/lib/config.js#L40-L46

But for some strange reason the default is dev and not default itself.

**Comment by wzrdtales** [comment]
Or more specific, default is as long default as it is defined.

**Comment by toymachiner62** [comment]
I'm just thinking about this some more and what if we had an option in the database.json file to set the default environment to use when no `-e` or `--env` option is set? That way I can set it in my database.json to be undefined if that's what I want, you can set it to dev, and the next guy can set it to local if they want.

Thoughts?

**Comment by wzrdtales** [comment]
That would be a no change, because you can already ;)

Everything that is a parameter is also a config option.

**Comment by toymachiner62** [comment]
Oh really? How would I specify that in the database.json file? Just with `env: 'default'`?

**Comment by wzrdtales** [comment]
Oh well, I see. I'm in fault, everything except env. Will add this though.

**Comment by toymachiner62** [comment]
Awesome. I like this solution. Thanks.

**Comment by wzrdtales** [comment]
ah well, ok. But is already there :D

`default: 'data'`

**Comment by toymachiner62** [comment]
what do you mean? So by specifying "default" that actually means the "env"?

**Comment by wzrdtales** [comment]
yes

**Comment by toymachiner62** [comment]
Oh well can we change that so it matches the params that can be passed in to the command line? I think that's much more intuitive.

**Comment by wzrdtales** [comment]
Not change but add it also to the list. Don't want to break any behavior, that is already in use.

**Comment by wzrdtales** [comment]
Have added it so far, at my local changes.

**Comment by toymachiner62** [comment]
What do you mean? You've added it to the list for version 0.9.0?

**Comment by toymachiner62** [comment]
oh i think i see so you can now specify either "default" or "env" in the database.json file and they'll do the same thing.

**Comment by wzrdtales** [comment]
y

**Comment by wzrdtales** [comment]
It's included in the latest, commit.
https://github.com/kunklejr/node-db-migrate/commit/ce69a1f4573aaf8d55071d0aadc975cd1fa77eb0

Is this issue for you resolved with this?
